### DISCUSSION TITLE
Should useSession rotate a new session when session is expired?

### DISCUSSION THREAD

**Discussion by ffleandro** [nextauthjs/next-auth#3166] [Help]
### Question 💬

I'm experiencing a weird behaviour and I'm not sure if I'm doing something wrong or expecting it wrongly.

I'm using jwt sessions and I've reduced the session `maxAge` to be really small (10 seconds) for testing purposes.

Whenever I navigate between next.js pages, everytime the `useSession` hook gets called I receive an expired session.

If I do a full browser refresh I now receive an `undefined` session instead of a new fresh session.

So my question is:

1) Should `useSession()` force a session rotation?
2) If I'm doing a full browser refresh should I receive an `undefined` session or should I get a fresh rotated session?

### How to reproduce ☕️

1) reduce jwt session maxAge to 10 seconds
2) navigate between pages and check the returned result from `useSession()`

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
The session is not refetched on clientside navigation. it just uses the one that is in the SessionProvider.

You can avoid session expiry by polling the session periodically https://next-auth.js.org/getting-started/client#refetch-interval

**Reply by ffleandro** [reply]
Is there an example where I can see client side session rotation?
